src/fileio/libhdfs_shim.cpp

```diff
@@ -346,22 +350,45 @@ extern  "C" {
     else return 0;
   }
 
+  static std::string get_hadoop_home_dir() {
+    static std::string hadoop_home = std::getenv("HADOOP_HOME_DIR");
+    return hadoop_home;
+  }
 
   static std::vector<fs::path> get_potential_libhdfs_paths() {
-    std::vector<fs::path> libhdfs_potential_paths = {
-      // find one in the unity_server directory
-      fs::path(turi::GLOBALS_MAIN_PROCESS_PATH + "/libhdfs.so"),
-      fs::path(turi::GLOBALS_MAIN_PROCESS_PATH + "/hdfs.dll"),
-      // find one in the local directory
-      fs::path("./libhdfs.so"),
-      fs::path("./hdfs.dll"),
-      // special handling for internal build path locations
-      fs::path(turi::GLOBALS_MAIN_PROCESS_PATH + "/../../../../deps/local/lib/libhdfs.so"),
-      fs::path(turi::GLOBALS_MAIN_PROCESS_PATH + "/../../../../deps/local/bin/hdfs.dll"),
-      // find a global libhdfs.so
-      fs::path("libhdfs.so"),
-      fs::path("hdfs.dll"),
+    static std::vector<fs::path> libhdfs_potential_paths = {
+
+      // Search order: 
+        // find one in the unity_server directory
+        // find one in the local directory
+        // Internal build path location; special handling there.
+        // Hadoop home dir
+        // Global search paths scoured by libhdfs.
+
+#ifdef __WIN32
+        fs::path(turi::GLOBALS_MAIN_PROCESS_PATH + "/hdfs.dll"),
+        fs::path("./hdfs.dll"),
+        fs::path(turi::GLOBALS_MAIN_PROCESS_PATH +
+                 "/../../../../deps/local/bin/hdfs.dll"),
+        fs::path(get_hadoop_home_dir() + "/lib/native/hdfs.dll"),
+        fs::path("hdfs.dll"),
+#elif __APPLE__
+        fs::path(turi::GLOBALS_MAIN_PROCESS_PATH + "/libhdfs.dylib"),
+        fs::path("./libhdfs.dylib"),
+        fs::path(turi::GLOBALS_MAIN_PROCESS_PATH +
+                 "/../../../../deps/local/lib/libhdfs.dylib"),
+        fs::path(get_hadoop_home_dir() + "/lib/native/libhdfs.dylib"),
+        fs::path("libhdfs.dylib")
```

Comment:
Include libhdfs.so even in the __APPLE__ path. Some hadoop build processes produce the shared library as "libhdfs.so".

Reply:
Okay, good to know.